soroban-env-host/src/vm.rs

```diff
@@ -102,6 +102,7 @@ impl Vm {
         if got_proto < want_proto {
             // Old protocols are finalized, we only support contracts
             // with similarly finalized (zero) prerelease numbers.
+            #[cfg(not(feature = "next"))]
```

Comment:
I think this deserves a comment explaining why do we compile this out in vnext.

Reply:
Good point. Added.